compiler/rustc_ty_utils/src/sig_types.rs

```diff
@@ -43,6 +43,12 @@ pub fn walk_types<'tcx, V: SpannedTypeVisitor<'tcx>>(
         // Walk over the type of the item
         DefKind::Static(_) | DefKind::Const | DefKind::AssocConst | DefKind::AnonConst => {
             if let Some(ty) = tcx.hir_node_by_def_id(item).ty() {
+                // If there are inference variables, we're gonna error out anyway, but
```

Comment:
```suggestion
                // If the type of the item uses `_`, we're gonna error out anyway, but
```
i was confused here. the point is that for `_` in item types we compute their type for diagnostics by running typeck using an infer var for the expected type, is it?

Reply:
yes, and then we error on them and tell the user what type to put there